web/client/utils/MapInfoUtils.js

```diff
@@ -89,7 +100,7 @@ const MapInfoUtils = {
     },
     buildIdentifyRequest(layer, props) {
         if (MapInfoUtils.services[layer.type]) {
-            return MapInfoUtils.services[layer.type].buildRequest(layer, props);
+            return MapInfoUtils.services[layer.type].buildRequest(layer, props, MapInfoUtils);
```

Comment:
???

Reply:
const MapInfoUtils = require('../MapInfoUtils'); in utils/mapinfo/wms.js returns an empty object.

Reply:
if (MapInfoUtils.services[layer.type]) {
            let infoFormat = MapInfoUtils.getDefaultInfoFormatValueFromLayer(layer, props);
            return MapInfoUtils.services[layer.type].buildRequest(layer, props, infoFormat);
        }

Reply:
This is because there is a dependency loop (`MapInfoUtils` includes `services/wms` too). This is usually a sign of bad design.
My suggestion is to move `getAvailableInfoFormat` and other WMS specific functionalities inside `wms.js`.